### DISCUSSION TITLE
Add Str::isHtml() method

### DISCUSSION THREAD

**Discussion by vaites** [laravel/framework#42678] [Ideas]
Something like `str($string)->isHtml()` can be useful. This can be implemented in both `Str` (as an static method) and `Stringable` classes just using `strip_tags` to avoid new dependencies. 

This method implementation was rejected twice before:

- [Pull Request 516](https://github.com/laravel/framework/pull/516): in 2013 @taylorotwell did not want to integrate it
- [Pull Request 37222](https://github.com/laravel/framework/pull/37222): wrong implementation

But I'm asking again because methods like `isJson()` have been implemented. I can send a PR with tests if you want, but don't want it to be rejected if you do not agree...

**Comment by bert-w** [comment]
hm they seem like handy functions, although I happen to never have needed them.

The implementation of that first PR for `isHtml` seems too simple though:
```
public static function isHtml($string)
{
    return strlen(strip_tags($string)) < strlen($string);
}
```
i.e. "anything with a tag is html". I can imagine that there will be a lot of edge cases since strictly speaking you have to abide by the whole HTML spec in order to judge it is valid. `<p><a>x</p></a>` is not valid HTML for instance.

**Comment by vaites** [comment]
I didn't want to add an strict validation of HTML (`<p><a>x</p></a>` is not valid but is HTML), just check if a string contains any tag. Web browsers _understands_ bad HTML, so this helper can do it too. 

Using `strip_tags` seems to be the right approach for this, and in the future an extra argument for stricter validation can be added.
